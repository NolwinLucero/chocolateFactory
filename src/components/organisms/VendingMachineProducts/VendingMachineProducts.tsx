import React from 'react';
import styled from 'styled-components';
import { Product } from '../../molecules/Product/Product';
import { ProductSchema } from '../../../schemas/product';

interface Props {
  products: ProductSchema[];
  credits: number;
  OnBuyHandler: (e: React.FormEvent, id: number) => void;
}

export const VendingMachineProducts: React.FC<Props> = ({
  products,
  credits,
  OnBuyHandler,
}) => {
  return (
    <>
      {products.map((product: ProductSchema, index: number) => (
        <StyledProductDiv key={index}>
          <Product
            product={product}
            credits={credits}
            OnBuyHandler={OnBuyHandler}
          />
        </StyledProductDiv>
      ))}
    </>
  );
};

const StyledProductDiv = styled.div`
  background-color: gold;
`;
