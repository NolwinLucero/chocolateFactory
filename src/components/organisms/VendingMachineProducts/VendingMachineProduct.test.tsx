import React from 'react';
import ReactDOM from 'react-dom';
import { VendingMachineProducts } from './VendingMachineProducts';
import renderer from 'react-test-renderer';

const credits = 5;
const products = [
  {
    id: 1,
    name: 'Caramel',
    price: '$3.10',
    stock: 3,
  },
  {
    id: 2,
    name: 'Hazelnut',
    price: '$2.50',
    stock: 3,
  },
  {
    id: 3,
    name: 'Organic Raw',
    price: '$2.00',
    stock: 3,
  },
];

const OnBuyHandler = () => {
  alert('clicked');
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <VendingMachineProducts
      products={products}
      credits={credits}
      OnBuyHandler={OnBuyHandler}
    ></VendingMachineProducts>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <VendingMachineProducts
        products={products}
        credits={credits}
        OnBuyHandler={OnBuyHandler}
      ></VendingMachineProducts>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
