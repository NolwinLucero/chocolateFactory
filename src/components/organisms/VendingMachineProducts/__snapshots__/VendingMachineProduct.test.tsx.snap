// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="sc-gsDKAQ buhTwb"
  >
    <div
      className="sc-bdvvtL bbdKiI"
    >
      <div>
        Caramel
      </div>
      <img
        alt="Caramel"
        src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Gooey-Chocolate-Caramel-Bars_EXPS_CMZ18_10572_D10_26_5b-1.jpg?fit=200,200"
      />
      <div>
        Stock: 
        3
      </div>
      <div>
        $3.10
      </div>
      <button
        data-testid="1"
        disabled={false}
        id="1"
        onClick={[Function]}
      >
        Buy
      </button>
    </div>
  </div>,
  <div
    className="sc-gsDKAQ buhTwb"
  >
    <div
      className="sc-bdvvtL bbdKiI"
    >
      <div>
        Hazelnut
      </div>
      <img
        alt="Hazelnut"
        src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Gooey-Chocolate-Caramel-Bars_EXPS_CMZ18_10572_D10_26_5b-1.jpg?fit=200,200"
      />
      <div>
        Stock: 
        3
      </div>
      <div>
        $2.50
      </div>
      <button
        data-testid="2"
        disabled={false}
        id="2"
        onClick={[Function]}
      >
        Buy
      </button>
    </div>
  </div>,
  <div
    className="sc-gsDKAQ buhTwb"
  >
    <div
      className="sc-bdvvtL bbdKiI"
    >
      <div>
        Organic Raw
      </div>
      <img
        alt="Organic Raw"
        src="https://www.tasteofhome.com/wp-content/uploads/2018/01/Gooey-Chocolate-Caramel-Bars_EXPS_CMZ18_10572_D10_26_5b-1.jpg?fit=200,200"
      />
      <div>
        Stock: 
        3
      </div>
      <div>
        $2.00
      </div>
      <button
        data-testid="3"
        disabled={false}
        id="3"
        onClick={[Function]}
      >
        Buy
      </button>
    </div>
  </div>,
]
`;
